recipes/ade/all/test_package/CMakeLists.txt

```
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package CXX)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
```

You need to set explicitly set the c++ standard for the test package.
Otherwise it will use the default one, which might be lower then the required one.
I've set it to c++11 as you tested for in `validate` (and is also written [here](https://github.com/opencv/ade#prerequisites-and-building))
```suggestion
target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
set_property(TARGET ${PROJECT_NAME} TARGET PROPERTY CXX_STANDARD 11)
```